CONTRIBUTING.md

```diff
@@ -1,21 +1,132 @@
-## Guidelines
+## Contributing Guidelines
 
-- Add one change per one commit.
+### Development environment
+Mimesis does not have any need for third-party tools, but we use a lot of tools in development stage,
+which you should install on your system if you want to contribute.
 
-- Always comment your code (only in English!).
+We use `Pipenv` for managing development packages:
+```
+~ pip install pipenv
+~ pipenv install --dev
+```
 
-- Check your spelling and grammar.
+### Code Style
+1. Every contributor must follow the [PEP8](https://www.python.org/dev/peps/pep-0008/) code style.
+2. Always use single-quoted strings, unless a single-quote occurs within the string.
+3. Always run tests, because tests check code style too (`pytest-flake8`).
 
-- Check code with [pycodestyle](https://github.com/PyCQA/pycodestyle) or with any other similar tool.
+### Annotating
+We are using optional static typing (`mypy`) and this means that every contributor should
+annotate his functions and methods (not variables, because Python 3.5 does not support annotation of variables).
 
-- Run the tests after each commit.
+```python
+def plus(a: int, b: int) -> int:
+    """Get sum of a and b.
 
-- Make sure the tests pass.
+    :param a: First number.
+    :param b: Second number.
+    :return: Sum of a and b.
+    """
+    return a + b
+```
 
-- If you add any functionality, then you should add tests for it.
+### Documenting
+Always add docstrings for your functions, methods, and classes:
 
-- Annotate your code.
+```python
+class Example(object):
+    """Example class which was created to show how we
+    document our classes, methods, and functions.
+    """
 
-- Do not write bad code!
+    def foo(self) -> str:
+        """This method always returns ``foo``.
+
+        :return: String ``foo``
+        """
+        return 'foo'
+
+def pi() -> float:
+    """Get Pi with two digits after the decimal point.
+
+    :return: Pi
+    """
+    return 3.14
+```
+
+It's a good idea to add a comment to code line where what is happening is not clear from the context,
+but you shouldn't add comments to pretty obvious things.
+
+### Documentation
+If you have added a new class, module or function than ones should be added to the `docs/api.rst`.
+
+### Testing
+You should write the test which shows that the bug was fixed or that the feature works as expected,
+run test before you commit your changes to the branch and create PR.
+
+To run tests, simply:
+```
+make test
+# or
+~ py.test --benchmark-skip --color=yes ./
+```
+
+Check out logs of Travis CI or AppVeyor if tests were failed on creating PR, there you can find useful information.
+
+### Performance
+It's good idea to run benchmark test, when you add your feature:
```

Comment:
Should we do it anyway? Maybe a CI job? 
Sometimes performance degradation is not obvious from the code.

Reply:
@sobolevn At this moment our benchmarks coverage only the default locale. I don't know should we or not.